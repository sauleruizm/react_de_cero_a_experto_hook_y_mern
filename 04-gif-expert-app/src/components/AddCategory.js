import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ( e ) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');            
        }
        //console.log('submit echo');
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Add category</h2>
            <p>{inputValue}</p>
            <input type="text" 
                value={inputValue} 
                onChange= { handleInputChange }/ >
        </form>
    )
    
}

AddCategory.propTypes =  {
    setCategories: PropTypes.func.isRequired
}