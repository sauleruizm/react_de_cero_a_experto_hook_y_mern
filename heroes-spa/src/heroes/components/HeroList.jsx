import { HeroCard } from './HeroCard'

import { getHeroesByPublisher } from '../helpers'
import { useMemo } from 'react'

export const HeroList = ({ publisher }) => {
	const heroes = useMemo( ()=> getHeroesByPublisher(publisher), [ publisher ] )

	return (
		<div className='row rows-cols-1 row-cols-md-3 g-3'>

			{heroes.map((hero) => {
				// console.table(hero)
				return <HeroCard key={hero.id}  {...hero} />
			})}
		</div>
	)
}
