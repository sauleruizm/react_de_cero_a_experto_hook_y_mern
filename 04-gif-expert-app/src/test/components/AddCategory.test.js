import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
    
    const setCategories = jest.fn();
    let wrapper;

    beforeEach( ()=> {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories= {setCategories} />);
    } );


    test('debe de mostrarse correctamente ', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);
    })


    test('no debe de postear la informaciÃ³n con submit', () => {
        
        wrapper.find('form').simulate('submit',{ preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();

    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        //simular el imput change
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value);

        //simular el submit
        wrapper.find('form').simulate('submit',{ preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();

        expect(wrapper.find('input').text().trim()).toBe('');
    })
    
    
})
