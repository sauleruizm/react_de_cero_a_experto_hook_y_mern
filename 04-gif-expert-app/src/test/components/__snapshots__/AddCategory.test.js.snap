// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add category
  </h2>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
